core/math/triangle_mesh.cpp

```diff
@@ -231,14 +231,12 @@ Vector3 TriangleMesh::get_area_normal(const AABB &p_aabb) const {
 			}
 			case VISIT_LEFT_BIT: {
 				stack[level] = (VISIT_RIGHT_BIT << VISITED_BIT_SHIFT) | node;
-				stack[level + 1] = b.left | TEST_AABB_BIT;
-				level++;
+				stack[++level] = b.left | TEST_AABB_BIT;
```

Comment:
That's pretty confusing that a compiler would trip up on this... But if it works `¯\_(ツ)_/¯``

Another option if it works (might be more friendly in the Godot codebase as we usually avoid the ambiguity of `++i` and `i++` (personally I need to think a few seconds each time to be sure about what's going to happen):
```suggestion
				level++;
				stack[level] = b.left | TEST_AABB_BIT;
```
Though I'm not strongly opposed to the current form either, in the context of this code it's fairly explicit.

Reply:
That doesn't appear to be related to this PR at all.

Reply:
This was mostly from a friend's suggestion and I actually did have it that way (which also fixed it), but honestly, I'm happy to do it whatever way you'd prefer. All I know is that the way it was being was causing bugs in MSVC

Reply:
Also, my bad about the other issue. That's clearly an oversight. I will correct it